src/mem/pool.h

```diff
@@ -32,11 +33,33 @@ namespace snmalloc
     constexpr PoolState() = default;
   };
 
-  template<typename T>
+  /**
+   * Class used to instantiate non-allocator pools using a Singleton PoolState.
+   */
+  template<typename T, typename SharedStateHandle>
+  class PoolStateHandle
+  {
+    static PoolState<T>& pool_state =
+      Singleton<PoolState<T>, PoolStateHandle::make_state>::get();
+
+    static PoolState<T>* make_state()
+    {
+      return ChunkAllocator::alloc_meta_data<PoolState<T>, SharedStateHandle>(
+        nullptr);
+    }
+
+  public:
+    static PoolState<T>& pool()
+    {
+      return pool_state;
+    }
+  };
+
+  template<typename T, typename SharedStateHandle>
```

Comment:
If you think all the uses of this are going to be of the shape `Pool<T<U>, U>`, maybe codify that?  `template<template<typename U> TC, U> class Pool { using T = TC<U>; /* ... */ }` ?

Reply:
I don't think that all uses will be of the form Pool<T<U>< U>. Only the allocator pool has this shape for now, the Verona pools (LocalEpoch and LocalLog) don't. But thanks for the comment, I will try to keep this pattern in mind for the future.